clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -61,7 +61,7 @@ public static void registerAppInfo(String prefix, String id) {
         }
     }
 
-    public static void unregisterAppInfo(String prefix, String id) {
+    public static synchronized void unregisterAppInfo(String prefix, String id) {
```

Comment:
If this must be `synchronized`, don't we have to do the same for `registerAppInfo`?